web/app/js/components/ResourceDetail.jsx

```diff
@@ -154,15 +154,24 @@ export class ResourceDetailBase extends React.Component {
         // Do this by querying for metrics for all pods in this namespace and then filtering
         // out those pods whose owner is not this resource
         // TODO: fix (#1467)
-        let podBelongsToResource = _reduce(podListRsp.pods, (mem, pod) => {
-          if (_get(pod, resourceTypeToCamelCase(resource.type)) === resource.namespace + "/" + resource.name) {
-            mem[pod.name] = true;
-          }
+        let resourceName = resource.namespace + "/" + resource.name;
+        let podMetricsForResource;
+
+        if (resource.type === "pod") {
+          podMetricsForResource = _filter(podMetrics, pod => pod.name === resource.name);
+        } else {
+          let podBelongsToResource = _reduce(podListRsp.pods, (mem, pod) => {
+            if (_get(pod, resourceTypeToCamelCase(resource.type)) === resourceName) {
+              mem[pod.name] = true;
```

Comment:
realizing this code was already in place, but would you mind adding a comment about the structure of `pod.name`? something like...
```diff
diff --git a/web/app/js/components/ResourceDetail.jsx b/web/app/js/components/ResourceDetail.jsx
index 6744dbdd..cccfc054 100644
--- a/web/app/js/components/ResourceDetail.jsx
+++ b/web/app/js/components/ResourceDetail.jsx
@@ -157,21 +157,24 @@ export class ResourceDetailBase extends React.Component {
         let resourceName = resource.namespace + "/" + resource.name;
         let podMetricsForResource;

+        // note pod.name in podMetrics != pod.name in podListRsp
         if (resource.type === "pod") {
+          // pod.name == emoji-85977956cf-5v46h
           podMetricsForResource = _filter(podMetrics, pod => pod.name === resource.name);
         } else {
           let podBelongsToResource = _reduce(podListRsp.pods, (mem, pod) => {
             if (_get(pod, resourceTypeToCamelCase(resource.type)) === resourceName) {
+              // pod.name == emojivoto/emoji-85977956cf-5v46h
               mem[pod.name] = true;
             }

             return mem;
           }, {});

+          // pod.name == emoji-85977956cf-5v46h
           podMetricsForResource = _filter(podMetrics, pod => podBelongsToResource[pod.namespace + "/" + pod.name]);
         }

-
         let resourceIsMeshed = true;
         if (!_isEmpty(this.state.resourceMetrics)) {
           resourceIsMeshed = _get(this.state.resourceMetrics, '[0].pods.meshedPods') > 0;
```